server/segment_rpc.go

```diff
@@ -330,6 +332,11 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 		return nil, err
 	}
 
+	// update OrchestratorInfo if necessary
+	if tr.Info != nil {
```

Comment:
Will this field only be populated when OrchestratorInfo needs to be updated?

Reply:
Yep. So, under normal circumstances `tr.Info` will be nil and it will only be non-nil when updated information needs to be returned. This is also why `OrchestratorInfo` is assigned a higher field number > 15 within the `TranscodeResult` message since [field numbers 1-15 generally should be reserved for very frequently occurring message elements](https://developers.google.com/protocol-buffers/docs/proto#assigning-field-numbers)

Reply:
That makes sense. Great attention to detail with `OrchestratorInfo` being higher than 15 here